### PR TITLE
Synthetic tests for prow-imported jobs

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-268](https://issues.redhat.com//browse/TRT-268)

This significantly refactors synthetic test processing, and makes it work for Prow.  This was a difficult refactor, since the original code was so tightly coupled to test grid.

- Extract synthetic test processing to be as generic as possible
- Convert Prow test suites to a RawJobRunResult (which contains OpenShift-specific stuff like operator, upgrade, install, etc)
- Creates synthetic tests based on that result
- Calculate overall job result based on the RawJobRunResult

Todo's:
- [x] Spot check prow results to confirm correctness
- [x] Confirm results are still the same for TestGrid job runs
- [x] Monitor memory usage during fetching to see impact
- [x] Improve efficiency of the conversion to RawJobRunResult (my test handling is weird from DB)

### CODE REVIEW

**Comment by stbenjam** [comment]
@dgoodwin This should be ready for a review.  In my testing, testgrid results are similar between this PR (left) and production (right).  Small difference is due to slight difference in age of the testgrid data.

image

File: pkg/prowloader/testconversion/testconversion.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	testgridv1 "github.com/openshift/sippy/pkg/apis/testgrid/v1"
+	"github.com/openshift/sippy/pkg/db/models"
+	"github.com/openshift/sippy/pkg/synthetictests"
+	"github.com/openshift/sippy/pkg/testidentification"
+)
+
+func ConvertProwJobRunToSyntheticTests(pj prow.ProwJob, tests map[string]*models.ProwJobRunTest, manager synthetictests.SyntheticTestManager) (*junit.TestSuite, v1.JobOverallResult) {
+	jrr := v1.RawJobRunResult{
+		Job:       pj.Spec.Job,
+		Failed:    pj.Status.State != prow.SuccessState,
```
Comment (dgoodwin) [inline pkg/prowloader/testconversion/testconversion.go:16]: Even if filtered elsewhere this looks a little risky, Pending and Triggered could both trigger as "failed" here. I'd go more explicit. Is aborted a failure as well? (probably yes for our purposes)

**Review by dgoodwin — COMMENTED** [review]
This is about all I can see. I'd say we're ok to try it out.

File: pkg/prowloader/testconversion/testconversion.go (hunk 1 above)
Comment (stbenjam) [reply pkg/prowloader/testconversion/testconversion.go]: Fair enough, fixed in last commit.

**Pull request merged**